cernopendata_client/downloader.py

```diff
@@ -29,26 +37,49 @@ def show_download_progress(download_t, download_d, upload_t, upload_d):
     STREAM.flush()
 
 
-def download_single_file(path=None, file_location=None):
+def download_single_file(path=None, file_location=None, protocol=None):
     """Download single file."""
-    file_name = file_location.split("/")[-1]
-    file_dest = path + "/" + file_name
-    with open(file_dest, "wb") as f:
-        display_message(
-            prefix="single",
-            msg="File: ./{}/{}".format(
-                path,
-                file_name,
-            ),
-        )
-        c = pycurl.Curl()
-        c.setopt(c.URL, file_location)
-        c.setopt(c.WRITEDATA, f)
-        c.setopt(c.NOPROGRESS, False)
-        c.setopt(c.XFERINFOFUNCTION, show_download_progress)
-        c.perform()
-        c.close()
-    print()
+    if protocol == "http":
+        file_name = file_location.split("/")[-1]
+        file_dest = path + "/" + file_name
+        with open(file_dest, "wb") as f:
+            display_message(
+                prefix="single",
+                msg="File: ./{}/{}".format(
+                    path,
+                    file_name,
+                ),
+            )
+            c = pycurl.Curl()
+            c.setopt(c.URL, file_location)
+            c.setopt(c.WRITEDATA, f)
+            c.setopt(c.NOPROGRESS, False)
+            c.setopt(c.XFERINFOFUNCTION, show_download_progress)
+            c.perform()
+            c.close()
+        print()
+    elif protocol == "root":
+        if not xrootd_available:
+            display_message(
+                prefix="double",
+                msg_type="error",
+                msg="xrootd is not installed on system.",
```

Comment:
You can also add to the error message "Please use the http protocol instead." to inform the users about what they should do instead.

```
==> ERROR: XRootD is not installed on system. Please use the 'http' protocol instead.
```

Reply:
Done 👍